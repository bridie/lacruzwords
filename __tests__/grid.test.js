import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import { currentTheme, sampleActiveCells, sampleInactiveCells } from './fixtures/currentTheme';	
import Grid from './../src/components/grid';

function setup() {
	const props = { currentTheme }

	const enzymeWrapper = shallow(<Grid {...props} />)

	return {
		props,
		enzymeWrapper
	}
}

describe("Grid Component", () => {
	it('renders correctly', () => {
		const { props } = setup();

  		const grid = renderer.create(<Grid {...props} />).toJSON();
  		expect(grid).toMatchSnapshot();
	});

	it('correctly specifies whether a cell is active or not', () => {
		const { enzymeWrapper } = setup();

		sampleActiveCells.forEach(sampleActiveCell => {
			expect(enzymeWrapper.instance().isActive(...sampleActiveCell)).toEqual(true);
		});

		sampleInactiveCells.forEach(sampleInactiveCell => {
			expect(enzymeWrapper.instance().isActive(...sampleInactiveCell)).toEqual(false);
		});
	});
})

