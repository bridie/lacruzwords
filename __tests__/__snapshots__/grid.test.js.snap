// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Component renders correctly 1`] = `
<div
  className="grid"
>
  <div
    className="grid__row"
  >
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
  </div>
  <div
    className="grid__row"
  >
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
  </div>
  <div
    className="grid__row"
  >
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
  </div>
  <div
    className="grid__row"
  >
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
  </div>
  <div
    className="grid__row"
  >
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
  </div>
  <div
    className="grid__row"
  >
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
  </div>
  <div
    className="grid__row"
  >
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
  </div>
  <div
    className="grid__row"
  >
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
  </div>
  <div
    className="grid__row"
  >
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
  </div>
  <div
    className="grid__row"
  >
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
  </div>
  <div
    className="grid__row"
  >
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
  </div>
  <div
    className="grid__row"
  >
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--active"
      disabled={false}
      maxLength="1"
      type="text"
    />
    <input
      className="grid__cell grid__cell--inactive"
      disabled="disabled"
      maxLength="1"
      type="text"
    />
  </div>
</div>
`;
